app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -964,13 +964,13 @@ public function getShippingAndTaxEstimate( $quote )
                 );
             }
 
-            $newDiscountedPrice = $quote->getSubtotalWithDiscount();
+            $adjustedShippingAmount = $this->getAdjustedShippingAmount($originalDiscountedSubtotal, $quote);
 
             $option = array(
                 "service" => $label,
                 "reference" => $rateCode,
-                "cost" => round(($quote->getShippingAddress()->getShippingAmount() + ($originalDiscountedPrice - $newDiscountedPrice)) * 100),
```

Comment:
@johnmccombs1  Can you please review this code and see what the effect of changing the following line will yield?

from:
```
 "cost" => round(($quote->getShippingAddress()->getShippingAmount() + ($originalDiscountedPrice - $newDiscountedPrice)) * 100),
```

to:
```
 "cost" => round(($quote->getShippingAddress()->getShippingAmount() + ($newDiscountedPrice - $originalDiscountedPrice)) * 100),
```

Above, I switch the order of $newDiscountedPrice and $originalDiscountedPrice, which appears to be the source of the logical error.

Perhaps, this one change can be substituted for all the other code.

Reply:
I've tested this and you are right. Not sure how I didn't see that as a fix before. Great find! I'll remove my code and add a unit test and call it done. Thanks!

Reply:
@johnmccombs1 LGTM